src/air/public_memory.cairo

```diff
@@ -43,6 +43,48 @@ impl PageImpl of PageTrait {
             i += 1;
         }
     }
+
+    fn extract_range(self: @Page, addr: felt252, len: felt252) -> Span<felt252> {
+        let mut arr = ArrayTrait::new();
+        let mut i = 0;
+
+        loop {
+            if i == len {
+                break arr.span();
+            }
+
+            let current = *self.at((addr + i).try_into().unwrap());
+
+            // TODO is this needed? If not we can just use slice directly 
```

Comment:
this check is needed
when u have multiple memory pages this is going to fail because indexes will lose their consistency so that is cool

Reply:
but given we have other assertion for single memory page we could omit this one

Reply:
i would leave it as is rn, we can keep in mind that this can probably be optimized
we could have function mem_check_range for this kind of mem consistency checks

Reply:
Makes sense, I agree let's leave it as it is